proxy/build.rs

```diff
@@ -24,4 +24,11 @@ fn build_control() {
         .enable_server(true)
         .build(server_files, dirs)
         .unwrap();
+
+    // recompile protobufs only if any of the proto files changes...
+    for file in dirs.iter().chain(client_files).chain(server_files) {
```

Comment:
Is `dirs.iter()` needed here? It seems like it isn't needed and it could be just `client_files.iter().chain(server_files)`.

Reply:
The [documentation](https://doc.rust-lang.org/cargo/reference/build-scripts.html#outputs-of-the-build-script) recommends outputting the name of the directory as well as the files within it, but this could be removed.

Reply:
> The documentation recommends outputting the name of the directory as well as the files within it, but this could be removed.

The documentation doesn't say exactly that. It's not wrong the way it is but IMO it would be more right to not include the directory. (Originally this feature did do something more useful when a directory was listed, however it was found to be broken, and so the feature basically doesn't do anything useful when a directory is listed.)